### DISCUSSION TITLE
How are feed updates handled?

### DISCUSSION THREAD

**Discussion by Panja0** [FreshRSS/FreshRSS#5012] [Q&A]
I'm using a cron job to do a web update (curl 127.0.0.1/i/etcetcetc) for 1 particular feed.
This feeds get a lot of new articles and needs to be updated frequently so I've set the cron job for every 2 minutes (*/2).
But articles can sometimes take 10 - 20 minutes before they show up in my FreshRSS instance.

Is this normal? If so can it be changed?

**Comment by math-GH** [comment]
You can set the default minimum time in the config (minimum 20 minutes) :
![image](https://user-images.githubusercontent.com/1645099/212132168-76fe5cea-5108-4f12-807c-279e3ae141cb.png)

In the feed settings itself you can change it too (minimum: 15 minutes):
![image](https://user-images.githubusercontent.com/1645099/212132471-b609ab5a-6810-4041-a27c-4f50a9790661.png)

**Comment by Panja0** [comment]
Thanks for the reply. 

But what does the web update via cron do than? I’ve set that to every 2 minutes. 

This particular feeds needs regular updating. Once per 15 min won’t work in this case.

**Comment by math-GH** [comment]
The cron checks the feeds, whether they should be updated next (check last update date and calculate the next refresh date).

If there is no feed to be updated, then the cron will do nothing

**Comment by Panja0** [comment]
Ok. How would I proceed if I want to check one particular feed every 2 minutes?

**Comment by math-GH** [comment]
What is your use case to fetch every 2 minutes a RSS feed?

**Comment by Panja0** [comment]
I’m using Full-Text RSS to get the full article from the RSS feed. Though the feed limits full articles to 2 articles per time. They post regular updates, sometimes a few per minute. 

There for I need faster updates otherwise I’ll lose articles. Does that make sense?

**Reply by Alkarex** [reply]
RSS feeds typically contain the last 10, 50, or even 100 published articles, so you typically do not risk loosing any article by refreshing at a slower rate. See my full answer below.

**Comment by Alwaysin** [comment]
If I'm not mistaken, FreshRSS has a hard-coded minimum value for feeds retrieval so as not to harm website owners (who can then block tools like FreshRSS if they see too many requests, which is bad for everyone !). If you want to change it, you'll have to manually update the source code.

I think for this use case it is best to use WebSub which is supported by FreshRSS. Maybe you could talk about it to the site owner?

**Comment by Alkarex** [comment]
FreshRSS does not, by design, supports pull refreshes at frequencies higher than once every 15 minutes. But FreshRSS supports instant push.

https://github.com/FreshRSS/FreshRSS/issues/2262#issuecomment-470060813
> We are part of an ecosystem. A typical reaction that we have seen from several servers is to simply ban by user-agent (that is why we had to remove "SimplePie" from our user-agent string, because it was too frequently banned), or to remove their RSS feed altogether. Bad user behaviours affect the larger community.

> If you need (near-)realtime updates, use the WebSub push mechanism, which is exactly meant for that, and is very easy to deploy server-side. FreshRSS has built-in client support for it.

https://github.com/FreshRSS/FreshRSS/issues/2595#issuecomment-546416937
> Under the settings for individual feeds, you can go down to 15min.
The lowest global/default purposely cannot be set faster than every 20 minutes, to avoid wasting resources and make sure the RSS ecosystem remains sane.
We have regularly seen some sites blocking by IP, or worse, by user agent, due to abuses with too frequent refreshes. 
For faster / near instant updates, I suggest looking into the WebSub protocol (formerly PubSubHubbub), which is supported by FreshRSS. For that to work, your server needs to be routable (visible from Internet), in order to receive callbacks as soon as there is something new. 
I hope this clarifies the situation 🙂

For niche cases, you can consider using a bridge for a specific feed, e.g. [RSS Bridge](https://rss-bridge.github.io/rss-bridge/)

**Comment by Panja0** [comment]
WebSub would be awesome but the site does not support this. Typical RSS feeds indeed do contain 10 or more feeds but this particular site does not. There for to function properly I need a faster refresh than normal. If FreshRSS can’t do that I will lose articles for sure. Pity it’s not supported. Thanks for the help though.

**Reply by Alkarex** [reply]
For specific cases like that, you can consider using a bridge (check e.g. RSS Bridge). But refreshing that often is not a good design and is harmful for the RSS ecosystem. So contacting the Web site owners to hint them at some potential technical improvements would be best.
